modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/near/consistency/GridNearReadRepairAbstractFuture.java

```diff
@@ -344,21 +352,24 @@ protected final void recordConsistencyViolation(
 
             for (KeyCacheObject key : fut.keys()) {
                 if (inconsistentKeys.contains(key)) {
-                    Map<ClusterNode, CacheConsistencyViolationEvent.EntryInfo> map =
-                        entries.computeIfAbsent(
-                            ctx.unwrapBinaryIfNeeded(key, !deserializeBinary, false, null), k -> new HashMap<>());
+                    sensitiveKeyMap.computeIfAbsent(key, k -> includeSensitive
+                        ? ctx.unwrapBinaryIfNeeded(k, !deserializeBinary, false, null)
+                        : "[HIDDEN_KEY#" + UUID.randomUUID() + "]");
```

Comment:
Why do we need random UUID here?

Reply:
We need some unique key identification, otherwise, we'll be unable to mention more than one key per event.
UUID seems to be a good choice.
Other approaches like incremental id will lead to some inconvenience, like the same id at different events or the necessity to have global incremental value.

Reply:
> UUID seems to be a good choice.

Let's have `Arrays.hashCode(keyBytes)` for both "HIDDEN_KEY" and "HIDDEN_VALUE" logging.

Reply:
This may help to guess the data, which is unacceptable.